### PR TITLE
feat(framework) Refactor `flwr ls` table formatter

### PR DESCRIPTION
**Pull request by chongshenng**
Split the logic of formatting the `Run` from creating a `rich.Table`. The formatted `Run` will be used later to create a JSON output.

Merge before:
* #4612

### CODE REVIEW

File: src/py/flwr/cli/ls.py [hunk 1]
```diff
@@ -139,13 +139,15 @@ def _init_channel(app: Path, federation_config: dict[str, Any]) -> grpc.Channel:
     return channel
 
 
-def _format_run_table(run_dict: dict[int, Run], now_isoformat: str) -> Table:
-    """Format run status as a rich Table."""
+def _format_run(run_dict: dict[int, Run], now_isoformat: str) -> list[tuple[str, ...]]:
+    """Format run status to a list."""
```
Comment (panh99) [inline src/py/flwr/cli/ls.py:143]:
```suggestion
def _format_runs(run_dict: dict[int, Run], now_isoformat: str) -> list[tuple[str, ...]]:
    """Format runs to a list."""
```

File: src/py/flwr/cli/ls.py [hunk 2]
```diff
@@ -192,15 +194,60 @@ def _format_datetime(dt: Optional[datetime]) -> str:
...
+                _format_datetime(pending_at),
+                _format_datetime(running_at),
+                _format_datetime(finished_at),
+            )
         )
+    return run_status_list
+
+
+def _to_table(run_list: list[tuple[str, ...]]) -> Table:
+    """Format run status list to a rich Table."""
```
Comment (panh99) [inline src/py/flwr/cli/ls.py:214]:
```suggestion
    """Format the provided run list to a rich Table."""
```

File: src/py/flwr/cli/ls.py [hunk 3]
```diff
@@ -192,15 +194,60 @@ def _format_datetime(dt: Optional[datetime]) -> str:
...
+                f"{run.fab_version}",
+                f"{run.fab_hash}",
+                f"[{status_style}]{status_text}[/{status_style}]",
+                format_timedelta(elapsed_time),
+                _format_datetime(pending_at),
+                _format_datetime(running_at),
+                _format_datetime(finished_at),
+            )
         )
+    return run_status_list
```
Comment (panh99) [inline src/py/flwr/cli/ls.py:210]:
```suggestion
        run_list.append(
            (
                f"{run.run_id}",
                f"{run.fab_id}",
                f"{run.fab_version}",
                f"{run.fab_hash}",
                f"[{status_style}]{status_text}[/{status_style}]",
                format_timedelta(elapsed_time),
                _format_datetime(pending_at),
                _format_datetime(running_at),
                _format_datetime(finished_at),
            )
        )
    return run_list
```

**Review by panh99 — COMMENTED** [review]
Looks good! Just a few minor comments.

**Review by panh99 — APPROVED** [review]
LGTM!

**Review by danieljanes — APPROVED** [review]

**Pull request merged**